File: runinator-command-center/src/core/domain/models/auth/api-key.ts
export interface ApiKey {
  id: string | null;
  name: string;
  principal_kind: "user" | "service";
  principal_id: string;
  system_role?: "engine" | "worker" | "waker" | "agent" | "replica" | null;
  org_id?: string | null;
  action_ceiling: string[];
  key_prefix: string;
  last_used_at?: string | null;
  expires_at?: string | null;
  disabled: boolean;
  created_at: string;
}

export interface CreateApiKeyResponse {
  api_key: ApiKey;
  secret: string;
}

File: runinator-models/src/auth/api_key.rs
#[allow(unused_imports)]
use super::*;

/// API key metadata in wire form. never carries the secret or its hash.
#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ApiKey {
    pub id: Option<Uuid>,
    pub name: String,
    #[serde(default)]
    pub principal_kind: PrincipalKind,
    pub principal_id: Uuid,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub system_role: Option<SystemRole>,
    #[serde(default, skip_serializing_if = "Option::is_none")]
    pub org_id: Option<Uuid>,
    #[serde(default)]
    pub action_ceiling: Vec<Action>,
    pub key_prefix: String,
    #[serde(default)]
    pub last_used_at: Option<DateTime<Utc>>,
    #[serde(default)]
    pub expires_at: Option<DateTime<Utc>>,
    pub disabled: bool,
    pub created_at: DateTime<Utc>,
}
